start.sh

```diff
@@ -1,2 +1,374 @@
+      --report-path-arch)
+        REPORT_PATH_ARCH="$argVal"
+      ;;
+      -h|--help|-help|--h)
+        show_help
+      ;;
+      -*)
+        local dlm=""
+        if [ -n "$argVal" ]; then dlm="="; fi
+        OTHER_PARAMS="${OTHER_PARAMS} $argKey$dlm$argVal"
+      ;;
+      *)
+        let NAMELESS_COUNTER=NAMELESS_COUNTER+1
+        if [ $NAMELESS_COUNTER -eq 1 ]; then
+          SDL_CORE="$argVal"
+        elif [ $NAMELESS_COUNTER -eq 2 ]; then
+          TEST_TARGET="$argVal"
+        fi
+      ;;
+    esac
+  done
+}
+
+check_arguments() {
+  # check presence of mandatory arguments
+  if [ -z $SDL_CORE ]; then
+    echo "Path to SDL binaries was not specified"
+    exit 1
+  fi
+  if [ -z $TEST_TARGET ]; then
+    echo "Test target was not specified"
+    exit 1
+  fi
+  # check if defined path exists
+  if [ ! -d $SDL_CORE ]; then
+    echo "SDL core binaries was not found by defined path"
+    exit 1
+  fi
+  if [ ! -d $TEST_TARGET ] && [ ! -f $TEST_TARGET ]; then
+    echo "Test target was not found by defined path"
+    exit 1
+  fi
+  # set default values
+  if [ -z "$REPORT_PATH" ]; then
+    REPORT_PATH=${ATF_PATH}/TestingReports
+  fi
+  if [ -z "$REPORT_PATH_ARCH" ]; then
+    REPORT_PATH_ARCH=${ATF_PATH}/TestingReportsArch
+  fi
+  # add '/' to the end of the path if it missings
+  if [ "${SDL_CORE: -1}" = "/" ]; then
+    SDL_CORE="${SDL_CORE:0:-1}"
+  fi
+  if [ "${TEST_TARGET: -1}" = "/" ]; then
+    TEST_TARGET="${TEST_TARGET:0:-1}"
+  fi
+}
+
+log() {
+  local str
+  for arg in "$@"
+  do
+    str="$str$arg"
+  done
+  echo -e $str
+}
+
+logf() {
+  log "$@" | tee >(sed "s/\x1b[^m]*m//g" >> ${REPORT_PATH_ARCH_TS}/${REPORT_FILE})
+}
+
+backup() {
+  log "Back-up SDL files"
+  for FILE in ${SDL_BACK_UP[*]}; do cp -n ${SDL_CORE}/${FILE} ${SDL_CORE}/_${FILE}; done
+}
+
+restore() {
+  log "Restoring SDL files from back-up"
+  for FILE in ${SDL_BACK_UP[*]}; do cp -f ${SDL_CORE}/_${FILE} ${SDL_CORE}/${FILE}; done
+}
+
+clean_backup() {
+  log "Cleaning up back-up SDL files"
+  for FILE in ${SDL_BACK_UP[*]}; do rm -f ${SDL_CORE}/_${FILE}; done
+}
+
+await() {
+  local TIMEOUT=$1
+  local PID=$2
+  local TIME_LEFT=0
+  while true
+  do
+    local TIMEOUT_EXPIRED=$(( $TIMEOUT < $TIME_LEFT ))
+    if ! kill -0 ${PID} &>/dev/null
+    then
+      echo "Done"
+      return 0
+    fi
+    if ! (( TIMEOUT_EXPIRED ))
+    then
+      TIME_LEFT=$(( TIME_LEFT + 1 ))
+      sleep 1
+    else
+      echo "Timeout ($TIMEOUT sec) expired. Force killing: ${PID} ..."
+      kill -s SIGKILL ${PID}
+      sleep 0.5
+      echo "Done"
+      return 0
+    fi
+  done
+}
+
+kill_sdl() {
+  local PROCESS_NAME=smartDeviceLinkCore
+  local PIDS=$(ps -ef | grep -e "^$(whoami).*$PROCESS_NAME" | grep -v grep | awk '{print $2}')
+  for PID in $PIDS
+  do
+    local PID_INFO=$(pstree -sg $PID | head -n 1 | grep -v docker)
+    if [ ! -z "$PID_INFO" ]; then
+      log "'$PROCESS_NAME' is running, PID: $PID, terminating ..."
+      kill -s SIGTERM $PID
```

Comment:
@dboltovskyi I would avoid correct shutdown process in script runner. It should be the responsibility of the script to shut down SDL (if script care about sdl ignition cycle), in case if script don't care about ignition cycle, SDL may be just killed

Reply:
@LuxoftAKutsan There could be an issues if process was not terminated gracefully.
E.g. if socket connection was accidentally broken it could lead to a problem when a new connection will be open with some delay.
Currently there is a timeout in `kill_sdl()` function of 5 sec before process force termination.
It seems would be enough for correct termination.